netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -151,8 +172,23 @@ protected Status statusFromFailedFuture(ChannelFuture f) {
             return NettyClientTransport.this.statusFromFailedFuture(f);
           }
         },
-        method, headers, channel, authority, negotiationHandler.scheme(), userAgent,
-        statsTraceCtx);
+        method, headers, channel, authority, negotiatorScheme, userAgent, statsTraceCtx);
+  }
+  
+  private void lifecycleManagerNotifyTerminated(String reason, Throwable t) {
+    lifecycleManagerNotifyTerminated(reason, Utils.statusFromThrowable(t));
+  }
+
+  private void lifecycleManagerNotifyTerminated(String detail, Status s) {
+    if (!notifyClosed.compareAndSet(false, true)) {
```

Comment:
Should this notify-once logic go into ClientTransportLifecycleManager instead?

Reply:
I consider duplicate notifies to be a bug, hence putting the onus on the caller rather than the callee.